ui/src/features/Editor/components/TopBar/components/ActionBar/components/Version/hooks.ts

```diff
@@ -1,44 +1,55 @@
-import { useCallback, useState } from "react";
+import { useCallback, useRef, useState } from "react";
 import { Doc } from "yjs";
 import * as Y from "yjs";
 
 import { useToast } from "@flow/features/NotificationSystem/useToast";
 import { useDocument } from "@flow/lib/gql/document/useApi";
 import { useT } from "@flow/lib/i18n";
+import type { YWorkflow } from "@flow/lib/yjs/types";
 
 export default ({
   projectId,
   yDoc,
+  onDialogClose,
 }: {
   projectId: string;
   yDoc: Doc | null;
+  onDialogClose: () => void;
 }) => {
   const {
     useGetProjectHistory,
     useGetLatestProjectSnapshot,
     useRollbackProject,
+    useGetPreviewProjectSnapshot,
   } = useDocument();
-
   const { history, isFetching } = useGetProjectHistory(projectId);
-
   const { projectDocument } = useGetLatestProjectSnapshot(projectId);
-
   const [selectedProjectSnapshotVersion, setSelectedProjectSnapshotVersion] =
     useState<number | null>(null);
-  const [openVersionChangeDialog, setOpenVersionChangeDialog] =
+
+  const previewDocRef = useRef<Y.Doc | null>(null);
+
+  const [openVersionConfirmationDialog, setOpenVersionConfirmationDialog] =
     useState<boolean>(false);
   const [isReverting, setIsReverting] = useState<boolean>(false);
-  const snapshotOrigin = "snapshot-rollback";
+  const [isLoadingPreview, setIsLoadingPreview] = useState<boolean>(false);
+  const [previewDocYWorkflows, setPreviewDocYWorkflows] =
+    useState<Y.Map<YWorkflow> | null>(null);
+  const snapshotOriginRollback = "snapshot-rollback";
+  const snapshotOriginPreview = "snapshot-preview";
```

Comment:
These don't add anything to readability, and only add more code. Either use directly, or pull them out into the `src/global-constants.ts`